src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Parser {
+
+    private String inputString;
+
+    public Parser(String inputString) {
+        setInputString(inputString);
+    }
+
+    public void setInputString(String inputString) {
+        this.inputString = inputString;
+    }
+
+    public String getString() {
+        return inputString;
+    }
+
+    public void process(TaskList tasks, Ui ui) throws EmptyTaskException, InvalidInstructionException,
+            EmptyCommandException {
+        String instruction = inputString.split(" ", 2)[0];
+        if (instruction.equals("list")) {
+            ui.display(tasks);
+            return;
+        }
+        if (inputString.split(" ", 2).length == 0) {
+            throw new EmptyCommandException();
+        }
+        boolean flag1 = !(Arrays.asList("list", "mark", "unmark", "todo", "event", "deadline", "delete")
```

Comment:
I found it a little hard to read the code here, perhaps insert some comments to explain what's going on?